src/components/BooleanPill/BooleanPill.props.ts

```diff
@@ -0,0 +1,13 @@
+import { BooleanPillConfig } from './BooleanPill.styles';
+import { AvatarAppearance } from '../Avatar/types';
+
+export type BooleanPillProps = {
+  text: string;
+  state?: 'default' | 'disabled' | 'selected';
+  isInverted?: boolean;
+  custom?: BooleanPillConfig;
+  avatar?:
+    | { appearance?: 'image'; image: string }
+    | { appearance: Exclude<AvatarAppearance, 'image'>; initials: string };
+  onChange?: (state: boolean) => void;
```

Comment:
I think, we should allow here to support all native react synthetic events `onClick`, 'onFocus', `onBlur` etc.
You can add here optional `tabIndex` prop type which will be needed to support `:focus` state in css.